processor/attributesprocessor/attributes_log_test.go

```diff
@@ -0,0 +1,455 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attributesprocessor
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/exporter/exportertest"
+	"go.opentelemetry.io/collector/internal/data/testdata"
+	"go.opentelemetry.io/collector/internal/processor/filterconfig"
+	"go.opentelemetry.io/collector/internal/processor/filterset"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+)
+
+// Common structure for all the Tests
+type logTestCase struct {
+	name               string
+	inputAttributes    map[string]pdata.AttributeValue
+	expectedAttributes map[string]pdata.AttributeValue
+}
+
+// runIndividualLogTestCase is the common logic of passing trace data through a configured attributes processor.
+func runIndividualLogTestCase(t *testing.T, tt logTestCase, tp component.LogsProcessor) {
+	t.Run(tt.name, func(t *testing.T) {
+		ld := generateLogData(tt.name, tt.inputAttributes)
+		assert.NoError(t, tp.ConsumeLogs(context.Background(), ld))
+		// Ensure that the modified `ld` has the attributes sorted:
+		sortLogAttributes(ld)
+		require.Equal(t, generateLogData(tt.name, tt.expectedAttributes), ld)
+	})
+}
+
+func generateLogData(logName string, attrs map[string]pdata.AttributeValue) pdata.Logs {
+	td := pdata.NewLogs()
+	td.ResourceLogs().Resize(1)
+	rs := td.ResourceLogs().At(0)
+	rs.Resource().InitEmpty()
+	rs.InstrumentationLibraryLogs().Resize(1)
+	ils := rs.InstrumentationLibraryLogs().At(0)
+	spans := ils.Logs()
```

Comment:
nit: use a word that isn't `span` since it leads me to believe that we are working on a trace instead of a log